scripts/results_generation/generate_reproductions.py

```diff
@@ -65,76 +72,17 @@ def main():
-                        [0, 0, 0, 1, 0, 0],
-                        [0, 0, 0, 0, 1, 0],
-                        [magnitude, magnitude, magnitude, magnitude, magnitude, 1],
-                    ]
-                )
-                target = np.matmul(target, translation_matrix)
-                initial_state = np.matmul(initial_state, translation_matrix)
-            elif j == 1:
-                # apply rotation on both initial and target references
-                angle_rad = np.random.uniform(-math.pi, math.pi)
-                rotation_matrix = np.array(
-                    [
-                        [np.cos(angle_rad), np.sin(angle_rad), 0, 0, 0, 0],
-                        [-np.sin(angle_rad), np.cos(angle_rad), 0, 0, 0, 0],
-                        [0, 0, 1, 0, 0, 0],
-                        [0, 0, 0, np.cos(angle_rad), np.sin(angle_rad), 0],
-                        [0, 0, 0, -np.sin(angle_rad), np.cos(angle_rad), 0],
-                        [0, 0, 0, 0, 0, 1],
-                    ]
-                )
-                target = np.matmul(target, rotation_matrix)
-                initial_state = np.matmul(initial_state, rotation_matrix)
-            elif j == 2:
-                # apply scaling on both initial and target references
-                scaling_factor = np.random.uniform(0, 2)
-                scaling_matrix = np.array(
-                    [
-                        [scaling_factor, 0, 0, 0, 0, 0],
-                        [0, scaling_factor, 0, 0, 0, 0],
-                        [0, 0, scaling_factor, 0, 0, 0],
-                        [0, 0, 0, scaling_factor, 0, 0],
-                        [0, 0, 0, 0, scaling_factor, 0],
-                        [0, 0, 0, 0, 0, scaling_factor],
-                    ]
-                )
-                target = np.matmul(target, scaling_matrix)
-                initial_state = np.matmul(initial_state, scaling_matrix)
-            elif j == 3:
-                # apply noise on initial reference (simulate fixed target)
-                noise_i = 0.2 * np.random.rand(6)
-                target = regression.prediction[-1, 1:]
-                initial_state = apply_noise(target, noise_i)
-            elif j == 4:
-                # apply noise on target reference (simulate fixed initial position)
-                noise_t = 0.2 * np.random.rand(6)
-                target = apply_noise(target, noise_t)
-                initial_state = regression.prediction[0, 1:]
-            elif j == 5:
-                # apply noise on both initial and target references
-                noise_b = 0.2 * np.random.rand(6)
-                target = apply_noise(target, noise_b)
-                initial_state = apply_noise(initial_state, noise_b)
+            # apply noise on both initial and target references
+            noise = np.random.normal(scale=std_dev)
+            noise_vector = noise * np.ones(np.shape(target))
+            target = apply_noise(target, noise_vector)
```

Comment:
Shouldn't we simply use `target = target + noise_vector`? Right now I think we multiply by the noise but it's harder to understand that adding noise, no?

Reply:
We would anyhow need to scale noise_vector by the magnitude of each joint (you can see the noise_vector as the percentage of which the joints will be scaled, for instance if noise_vector is [0.1, 0.1, 0.1, 0.1, 0.1, 0.1], then the references will be scaled about 10%)
`target = target + noise_vector` wouldn't be correct since target can be [80, 55, 48, -20, -180, 6] for instance

Reply:
Ok I see what you are going for. If you use the multiplication. IMO the noise is not uniform anymore: it depend on the configuration you test how "deformed" by the noise it would be. A joint at angle 0 will never move then, right? on the other hand with a noise 0.01 a joint at 179 deg will move much more than one at 1 deg. Did i understand this correctly :)?

IMO I think it's easier to interpret results if we add zero mean Gaussian noise with a specific standard deviation to all joints. I'm trying to see the result in the paper:


| Noise std (deg) | Reproductions | GMCC | Joint MAE |
| -------------- | :--------- | ----------: | :----------: |
|1 | 30  | x |  x |
|5 | 30  | x |  x |
|10 | 30  | x |  x |
|15 | 30  | x |  x |
|20 | 30  | x |  x |
|25 | 30  | x |  x |

For example. Does that make sense to you?

Reply:
Yes that's right, I agree that it would depend on the configuration. On the other side, I think that's adding the same amount of noise it's even more counter-intuitive. 
1. it would depend on the unit (e.g deg as you mention here in the table) 
2. the 6 joints don't have the same working range, a difference of 10 deg on one joint could be much more meaningful that the same one on another joint 
However, I am not against your point, I see what you mean. I am just trying to understand which formulation makes more sense

Reply:
Point by point:

1. It always depend on the unit even if you do 10% ;). I think it's always preferable to have something explainable in simple terms: "with joint angle expressed in degree with apply a Gaussian noise with standard deviation of 5, 10, 15, 20, 25" is very uderstandable. Saying "For each joint, we apply noise corresponding to 10% of the actual joint angle" means I don't know the noise value without knowing the joint angles and looking at the table I cannot interpret the results then.
2. I don't understand how? That's only valid if we consider a real robot with physical limitation but there is no reason that it wouldn't work on a robot that has no limitations right? Since we run in simulation then it should be good no?

Reply:
Yes I was mainly concerned by the physical interpretation, if we totally abstract it then I am good with it.
Let's go like that, I will send the new dataset soon :)